From 490d9808e9adf0d12e34032ec6b7c85578275ab2 Mon Sep 17 00:00:00 2001
From: Vasilis Liaskovitis <vliaskovitis@suse.com>
Date: Wed, 19 Jul 2017 16:51:29 +0200
Subject: xen: hold lock_device_hotplug throughout vcpu hotplug operations
Patch-mainline: Never, SUSE specific
References: bsc#1042422

Holding the lock avoids crashes when issuing mulitple hot-add
/ hot-remove operations to the same vCPU(s) back to back.

[js] upstream handled disable_hotplug_cpu, so drop the hunk

Signed-off-by: Vasilis Liaskovitis <vliaskovitis@suse.com>
Acked-by: Juergen Gross <jgross@suse.com>
---
 drivers/xen/cpu_hotplug.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/xen/cpu_hotplug.c
+++ b/drivers/xen/cpu_hotplug.c
@@ -10,10 +10,12 @@
 
 static void enable_hotplug_cpu(int cpu)
 {
+	lock_device_hotplug();
 	if (!cpu_present(cpu))
 		xen_arch_register_cpu(cpu);
 
 	set_cpu_present(cpu, true);
+	unlock_device_hotplug();
 }
 
 static void disable_hotplug_cpu(int cpu)
